src/monmove.cpp

```diff
@@ -1381,6 +1527,73 @@ bool monster::attack_at( const tripoint &p )
     return false;
 }
 
+bool monster::pickup_at( const tripoint &p, item_location &target )
+{
+    //If called through normal pathfinding, then clear looting effect
+    if( p == goal ) {
+        remove_effect( effect_looting );
+
+    }
+
+    //Item has moved...
+    if( p != target.position() ) {
+        return false;
+    }
+
+    inv.push_back( *target.get_item() );
+    item *stored_item = &inv.back();
+
+    //Get weight of the stack or item
+    units::mass weight = target->weight();
+    units::mass capacity = this->weight_capacity();
+
+    int amount_taken = 1;
+
+    int charges = target->charges;
+
+    if( charges >= 1 ) {
+        //Adjust volume and weight for units in a stack
+        units::mass weigh_each = weight / charges;
+        amount_taken = charges + ( weigh_each / capacity );
+
+        //Caps the amount taken to avoid taking large stacks.
+        //TODO: better way to cap the amount taken?
+        if( amount_taken > 2 ) {
+            amount_taken = 2;
+        }
+
+        target->mod_charges( -amount_taken );
+        inv.back().charges = amount_taken;
+
+        //If we've taken all the stack, let's remove the item
+        if( amount_taken == charges ) {
+            target.remove_item();
+        }
+    } else {
+        //Remove item from ground
+        target.remove_item();
```

Comment:
This branch is taken when `charges < 1`. `amount_taken` is not changed here, so it remains at 1. The next branch will lead to `inv.pop_back();`.

So basically you remove the item from the ground (here) and you also remove the item from the inventory (`pop_back` below). In other words: the item vanishes into thin air.